cabal-install/src/Distribution/Client/IndexUtils.hs

```diff
@@ -276,11 +276,11 @@ getSourcePackagesAtIndexState verbosity repoCtxt mb_idxState mb_activeRepos = do
         IndexStateTime ts0 -> do
             when (isiMaxTime isi /= ts0) $
                 if ts0 > isiMaxTime isi
-                    then warn verbosity $
-                                   "Requested index-state " ++ prettyShow ts0
-                                ++ " is newer than '" ++ unRepoName rname ++ "'!"
-                                ++ " Falling back to older state ("
-                                ++ prettyShow (isiMaxTime isi) ++ ")."
+                    then die' verbosity $
+                                   "Stopping this command as the requested index-state=" ++ prettyShow ts0
+                                ++ " is newer than (" ++ prettyShow (isiMaxTime isi)
+                                ++ "), the most recent state of '" ++ unRepoName rname
```

Comment:
taking in account #8086 maybe we could change it with something like:
```suggestion
                                ++ "), the local state of '" ++ unRepoName rname
```

maybe `HEAD` instead of `local`? maybe it would be less clear for newcomers